### ISSUE TITLE
Metrics / Stats

### ISSUE THREAD

**Issue by waywardmonkeys** [derekjw/fyrie-redis#8]
Would be nice to be able to get this data:
- Number of messages and rate
- Latency
- # of outstanding requests

For the first two, I'd like to be able to set up a callback or something so that I can pump data directly into the metrics library that we already use (https://github.com/codahale/metrics).

For the last one, I'd like a function to call to get that (if it is even useful...)

**Comment by derekjw** [comment]
I added some simple hooks that will allow you to calculate the metrics you want. It is currently only in master:

https://github.com/derekjw/fyrie-redis/commit/83584dc754603e8e485c7f525629ce9ce0cf6cac

If this is something that you think will work for you, I can port it into the 'akka-1.2' and 'akka-1.1' branches.

**Comment by derekjw** [comment]
A simple example of setting up some callbacks to print to the console every 10 requests/results:

``` scala
    r onRequest { (id, ms) =>
      if (id % 10 == 0) println(id.toString + " requests")
    }
    r onResult { (id, ms) =>
      if (id % 10 == 0) println(id.toString + " results")
    }
```

**Comment by derekjw** [comment]
I've merged this into 'akka-1.2' and 'akka-1.1'. Let me know if you need something a bit different.
